src/services/cart.js

```diff
@@ -0,0 +1,111 @@
+        await cart.save();
+        return cart
+    }
+    
+    async findAllCarts(){
+        const carts = await Cart.findAll();
+        if (carts===null){
+            throw new Error('There are no existing Carts');
+        }else{
+            return carts;
+        }
+    }
+    
+    async findCartbyId(id){
+        //find the Cart by his own Id
+        const cart = await Cart.findByPk(id,{ include: [CartItem] });
+        if (cart===null){
+            throw new Error('Cart not found');
+        }else{
+            return cart;
+        }
+    }
+
+    async findCartbyUserId(userData) {
+        //Find the Cart by the User Id that owns the cart. 
+        const cart = await Cart.findOne({where: { UserId: userData.UserId}});  
+        if (cart===null){
+            throw new Error('User cart not found');
+        }else{
+            const data = await this.findCartbyId(cart.id);
+            return data;
+        }
+    }
+    
+    
+    async updateCart(data){
+        //update the Cart Data based on the values given
+        await Cart.update(
+            {
+                status: data.status,
+                total: data.total,
+                date: new Date()
+            },
+            {
+                where: {id:data.id}
+            }
+        );
+        const cart = await Cart.findAll({where:{id:data.id}})
+        if (cart===null){
+            throw new Error('Cart not found');
+        }else{
+            return cart;
+        }
+    }
+
+
+
+    async addProductToCart(data){
+        // Verify if the item already exists in the Cart
+        const exists= await CartItem.findOne({
+            where:{
+                CartId: data.CartId, 
+                ProductId: data.ProductId
+            }
+        })
+        if (!exists){
+            //item doesn't exists in the cart -> building item and saving it on DB, update to Cart total and status
+            const item= await CartItem.build({
+                qty: data.qty,
+                CartId: data.CartId,
+                ProductId: data.ProductId
+            })
+            await item.save();
+            await Cart.increment({total: +(data.price*data.qty)},{where: {id: data.CartId} });
+            await Cart.update({status: "Buying"},{where: {id:data.CartId}});
+
+        }else{
+            //item already exists in the Cart -> update to Cart total and CartItem quantity info about the item
+            const difference=(data.qty-exists.qty);
+            await Cart.increment({total: +(data.price*difference)},{where: {id: data.CartId} });
+            await CartItem.increment({qty: +difference},{where: {id: exists.id}});
+        }
+        //update the date of modification on the cart
+        await Cart.update({date: new Date()},{where: {id:data.CartId}});
+        const cart = await Cart.findAll({where:{id:data.CartId}})
+        if (cart===null){
+            throw new Error('Cart not found');
+        }else{
+            return cart;
+        }
```

Comment:
this can be replace with 

 `const cart = await Cart.update({date: new Date()},{where: {id:data.CartId}});`

 and you would get the 'updated' cart. No need to do the query

Reply:
Ok, did some research and found that, that option must be
`const cart = await Cart.update({date: new Date()},{where: {id:data.CartId}, returning: true});`
to get the whole Cart back and is only supported on Postgres according to the Sequelize [doc](https://sequelize.org/api/v6/class/src/model.js~model#static-method-update), so sadly here it only returns me this
![image](https://github.com/user-attachments/assets/793ca0d4-a1c2-400c-86a6-cf728eccb7b5)
and without the `returning:true` is just return a 1 alone.

Reply:
oh! ok, sorry about that. I thought it was 'universal' with sequelize. That's ok. Thank you for testing it